up42/utils.py

```diff
@@ -300,6 +320,119 @@ def _plot_images(
     plt.show()
 
 
+def _map_images(
+    plot_file_format: List[str],
+    result_df: GeoDataFrame,
+    filepaths,
+    aoi=None,
+    show_images=True,
+    show_features=False,
+    name_column: str = "id",
+    save_html: Path = None,
+) -> folium.Map:
+    """
+    Displays data.json, and if available, one or multiple results geotiffs.
+
+    Args:
+        plot_file_format: List of accepted image file formats e.g. [".png"]
+        result_df: GeoDataFrame of scenes, results of catalog.search()
+        aoi: GeoDataFrame of aoi
+        filepaths: Paths to images to plot. Optional, by default picks up the last
+            downloaded results.
+        show_images: Shows images if True (default).
+        show_features: Show features if True. For quicklooks maps is set to False.
+        name_column: Name of the feature property that provides the Feature/Layer name.
+        save_html: The path for saving folium map as html file.
+    """
+
+    centroid = box(*result_df.total_bounds).centroid
+    m = folium_base_map(
+        lat=centroid.y,
+        lon=centroid.x,
+    )
+
+    df_bounds = result_df.bounds
+    list_bounds = df_bounds.values.tolist()
+    raster_filepaths = [
+        path for path in filepaths if Path(path).suffix in plot_file_format
+    ]
+
+    # Make sure the quicklooks images are equal to the number of features in the scenes
+    try:
+        assert len(list_bounds) == len(raster_filepaths)
+    except AssertionError:
+        logger.error(
+            f"The length of the imgaes {len(raster_filepaths)} is not equal to number of"
+            f"row in scenes {len(result_df)}"
+        )
+        raise
```

Comment:
Logger message is really nice, but not sure this should be raised, we can still display something even if its not all of them no? Did that come up as an issue that not all quicklooks were downloaded or just for precaution?

Reply:
It was more for precaution, I can also remove it? what do you think?